### PR TITLE
Feat: Multiple selection of media in gutenberg

### PR DESCRIPTION
**Pull request by dratwas**
In this PR I added the possibility to set multiple media when inserting to the block.
Related `gutenberg-mobile` PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/1331
To test:
At this moment we don't support `gallery block` so this change can not be tested without changes in `gutenberg-mobile` JS bundle. In the future, we will use this functionality to multiple selections of media for the gallery.

* set multiple prop to true in media-upload bloc
* add image/video block - multiple select should be enabled for selecting media from the device or WP Media
* the array of selected media should be passed to the callback function

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift [hunk 1]
```diff
@@ -373,38 +376,40 @@ extension GutenbergViewController: GutenbergBridgeDelegate {
...
-    func gutenbergDidRequestMediaFromDevicePicker(filter: WPMediaType, with callback: @escaping MediaPickerDidPickMediaCallback) {
+    func gutenbergDidRequestMediaFromDevicePicker(filter: WPMediaType, allowMultipleSelection: Bool, with callback: @escaping MediaPickerDidPickMediaCallback) {
         mediaPickerHelper.presentMediaPickerFullScreen(animated: true,
                                                        filter: filter,
                                                        dataSourceType: .device,
+                                                       allowMultipleSelection: allowMultipleSelection,
                                                        callback: {(asset) in
-                                                        guard let phAsset = asset as? PHAsset else {
-                                                            callback(nil, nil)
+                                                        guard let phAsset = asset as? [PHAsset] else {
```
Comment (pinarol) [inline WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift]: I know this is still a draft but just wanted to point out the need for some variable names updated as plural.

**Comment by pinarol** [comment]
I see that this is a draft but the [gutenberg-mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/1331) looks ready to review. Is this still work in progress? If yes, you might want to mark gutenberg-mobile PR as `Not Ready to Review` as well, because reviewers won't be able to test it anyway.

**Comment by dratwas** [comment]
Hi all, this PR is ready. Could you please make a CR?

**Comment by pinarol** [comment]
Interestingly CI fails with error:

```
...
^D^DCloning into '.'...
Warning: Permanently added the RSA host key for IP address '192.30.253.113' to the list of known hosts.

agent key RSA SHA256:oEdX/44m2Y1klY5Uq287qOWuWfJrhusv4e5ImKQ5Uhk returned incorrect signature type

warning: Could not find remote branch pull/12404 to clone.
fatal: Remote branch pull/12404 not found in upstream origin
Exited with code 128

```

@jtreanor maybe you know this, this PR uses another fork. Is this expected?

**Comment by jtreanor** [comment]
@pinarol Thanks for the heads up. It looks like the CI failure is a configuration issue on our part. A recent change doesn't seem to be working correctly. I will look into a solution for it and update here.

**Comment by pinarol** [comment]
thank you @jtreanor 👍

**Comment by pinarol** [comment]
Could you update your branch from latest develop? @dratwas After that CI should be working

**Pull request merged**